frontend/src/components/authentication/SignIn.tsx

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable no-magic-numbers */
+/* eslint-disable react/jsx-no-bind */
+import React, {useState } from 'react'
+import {connect} from 'react-redux'
+import {Dispatch} from 'redux'
+
+import { fetchUser } from '../../actions/userActions'
+import { ActionsOfUser, UserDetailsOnLogin } from '../../actions'
+import {useStyles} from './SignInStyles'
+
+import {CssBaseline, TextField} from '@material-ui/core'
+import {Button, Avatar, Typography, FormControlLabel, Checkbox, Link, Paper, Grid }  from '@material-ui/core'
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined'
+
+export function SignInSide() :React.ReactElement {
+	const initialFormData = Object.freeze({
+		email: '',
+		password: '',
+	})
+
+	const [formData, updateFormData] = useState(initialFormData)
+
+	const handleChange = (e:React.SyntheticEvent) => {
```

Comment:
**Please read this comment, and remember**
 - Arrow syntax `() => {}` is creating new function.
 - When you using arrow syntax in react component render - this function will be recreated **each** render stage of this component
 - If you provide such function into children component - this will cause re-render of child component.
 - So, creating functions in render - is the anti-pattern
 - Providing this functions into child component - is also anti-pattern.

 For you case, you need to use class-based components, and create methods for `handleChange` and `handleSubmit
`